source/frontend/src/features/properties/map/shared/detail/models.ts

```diff
@@ -0,0 +1,42 @@
+export interface IAgreementFormData {
+  project_name: string;
+  project_number: string;
+  road: string; //this is the product name without the #
+  ps_file_no: string;
+  completion_date: string;
+  deposit_amount: string | null;
+  purchase_price: string;
+  owners: [
+    {
+      lastname: string;
+      firstname: string;
+      organization_name: string;
+      incorporation_number: string;
+      reg_no: string;
+    },
```

Comment:
for the non poc it would be better if they are defined on a different interface and just referenced here. Same with agents and properties